template/9-design.md

```diff
@@ -22,40 +22,112 @@
 
 ## Backend
 
-*Decompose the MVP into functional blocks.*
+### Application Logic
+
+All application logic will be managed on the backend. The client will communicate the necessary information to retrieve the appropriate data it needs to display, and the backend will respond accordingly. For instance, the client will send its position to the backend, which will then provide the nearby posts.
+
+This separation allows the logic to remain flexible and reduces the need for frequent app updates to fix bugs, as these can be addressed on the backend without modifying the client side. Additionally, this approach keeps the frontend lightweight and focused on optimizing the user experience.
+
+
+### Database Interactions
+
+All interactions with the database will be managed through the backend. The frontend will not have direct access to the database; instead, it will obtain the necessary data by communicating with the backend API.
+
+This design choice is primarily for consistency and security reasons. By restricting database access to the backend, we ensure that we can control and predict the actions performed on the database at any given time. This level of control would not be possible if the client had direct access to the database, as multiple versions of the application could lead to inconsistent interactions with the database. This approach is further justified when considering potential adversarial behavior from the client, where it is crucial to prevent direct database access to maintain security.
+
+
+### Listenable
+
+Some features of Proxima require the client to react to external events. For example, when a user is invited to join a group, the client needs to be notified of this event in real time. This will be accomplished through listenable Firestore documents.
+
+When a listenable logic is required, the cloud function will create a temporary document in a specific collection that the client will listen to. To maintain consistency with our *Database Interaction* policy, the client will only have read access to these documents, and any writes will be performed through backend API calls. This ensures that the client is promptly notified of relevant events while preserving the integrity and security of the backend.
+
+
+### Framework
+
+The backend for Proxima will utilize the Firebase suite. Specifically, the database will be managed through Firestore and the media storage will be held on Firebase Storage solution. The backend API will be implemented using Firebase Cloud Functions, written in Node.js as required by the framework. These functions will be callable from the client. This setup provides a scalable and efficient infrastructure for managing the application's backend operations, ensuring reliability and performance.
```

Comment:
```suggestion
The backend for Proxima will utilize the Firebase suite. Specifically, the database will be managed through Firestore and the media storage will be held on Firebase Storage Solution. The backend API will be implemented using Firebase Cloud Functions, written in Node.js as required by the framework. These functions will be callable from the client. This setup provides a scalable and efficient infrastructure for managing the application's backend operations, ensuring reliability and performance.
```

Reply:
I'm not sure of the phrasing here. Are you talking about "Firebase Storage Solution" ? If not, could we  maybe rephrase the end of this sentence like this : "... will be held on the Firebase Storage solution"? What do you think?

Reply:
That's a very good point indeed !

After looking it up, I believe that the actual name is Firebase Cloud Storage.
Thus, I propose that I replace `Firebase Storage solution` by `Firebase Cloud Storage`. What do you think ?

Reply:
Yes perfect 👍🏼

Reply:
Nice thanks !

Done in d6a3993ed41a90094e1348304bbf9ee9be19edc0 :+1: